src/pudl/helpers.py

```diff
@@ -339,6 +339,227 @@ def is_doi(doi):
     return bool(re.match(doi_regex, doi))
 
 
+def full_timeseries_temporal_merge(
```

Comment:
Is there a better way for this higher level function to accept all these arguments?

Reply:
It looks like `full_timeseries_temporal_merge` isn't being called anywhere in the code right now. Is that expected? Was @cmgosnell using it somewhere else in the RMI / PPL code?

Reply:
I want to use it in line with what I was trying to do in [this PR](https://github.com/catalyst-cooperative/pudl/pull/1478). I assume we can scrap that PR and insert this new merge into `pudl.analysis.allocate_net_gen.associate_generator_tables`.

I can also imagine it being the right thing to use in `pudl.analysis.mcoe.mcoe()`. The first merge in there uses an arg `all_gens` to determine the left or right-ness of the merge but if we're using an monthly `pudl_out` the output will include a ton of 1/1/year generator records for those that don't show up in the monthly data.